### PR TITLE
[Yul] Fix data flow analyzer for function definitions.

### PR DESCRIPTION
**Pull request by chriseth**
The data flow analyzer did not remove variables outside of the current function and thus still considered them to be available.

### CODE REVIEW

File: libyul/optimiser/DataFlowAnalyzer.cpp [hunk 1]
```diff
@@ -84,13 +84,26 @@ void DataFlowAnalyzer::operator()(Switch& _switch)
 
 void DataFlowAnalyzer::operator()(FunctionDefinition& _fun)
 {
+	// Save all information. We might rather reinstantiate this class,
+	// but this could be difficult if it is subclassed.
+	map<string, Expression const*> value;
```
Comment (leonardoalt) [inline libyul/optimiser/DataFlowAnalyzer.cpp]: Will this code be changed to use YulString?

Comment (chriseth) [reply libyul/optimiser/DataFlowAnalyzer.cpp]: Yep, will have to.

**Comment by chriseth** [comment]
Updated. @leonardoalt

**Comment by leonardoalt** [comment]
Windows compilation is failing

**Comment by chriseth** [comment]
@leonardoalt this is a problem in develop (already fixed). Do you want me to rebase?

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**